meshroom/ui/qml/GraphEditor/GraphEditor.qml

```diff
@@ -815,59 +812,97 @@ Item {
             }
 
             // Nodes
-            Repeater {
+             Repeater {
                 id: nodeRepeater
 
                 model: root.graph ? root.graph.nodes : undefined
 
                 property bool loaded: model ? count === model.count : false
-                property bool dragging: false
+                property bool ongoingDrag: false
+                property bool updateSelectionOnClick: false
                 property var temporaryEdgeAboutToBeRemoved: undefined
 
+                function isNodeSelected(index: int) {
+                    return uigraph.nodeSelection.isRowSelected(index);
+                }
+
                 delegate: Node {
                     id: nodeDelegate
 
                     node: object
                     width: uigraph.layout.nodeWidth
 
                     mainSelected: uigraph.selectedNode === node
-                    selected: uigraph.selectedNodes.contains(node)
                     hovered: uigraph.hoveredNode === node
 
+                    selected: nodeRepeater.isNodeSelected(index);
```

Comment:
Good point, I did not realize we can actually use `ItemSelectionModel.hasSelection` for that specific purpose:
https://doc.qt.io/qt-6/qml-qtqml-models-itemselectionmodel.html#hasSelection-prop
*It will trigger property binding updates every time selectionChanged() is emitted, even though its value hasn't changed.*